src/Microsoft.Health.ControlPlane.CosmosDb/Features/Storage/Rbac/CosmosRole.cs

```diff
@@ -0,0 +1,41 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Microsoft.Health.ControlPlane.Core.Features.Rbac.Roles;
+using Microsoft.Health.CosmosDb.Features.Storage;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.ControlPlane.CosmosDb.Features.Storage.Rbac
+{
+    public class CosmosRole : Role
+    {
+        internal const string RolePartition = "Role_";
+
+        public CosmosRole(Role role)
+              : base(role.Name, role.ResourcePermissions,  role.Version)
+        {
+            ETag = $"\"{role.Version}\"";
```

Comment:
We probably don't actually need this since the version will always be set by CosmosDB.